src/middleware/authentication.js

```diff
@@ -0,0 +1,27 @@
+/* Validates the user-provided auth token */
+const authentication = (req, res, next) => {
+  const authHeader = req.headers.Authorization;
+
+  /* Overrides user role for local development and testing purposes */
+  if (process.env.NODE_ENV !== 'production') {
+    const { role = 'admin' } = req.query;
+    req.user = { role }; // TODO: Update the shape of this object to match Firebase's user object
+    return next();
+  }
+
+  if (authHeader) {
+    // eslint-disable-next-line
+    const token = authHeader.split(' ')[1];
+
+    // TODO: Verify the Firebase token is valid
```

Comment:
Here we want to verify the Firebase token to make sure that it's valid. We don't check it's user role here since that will get checked in `authorization.js`

If the token is valid, place the entire user object on the `req` object like `req.user` = `token.user`, or whatever makes sense.